src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -1220,7 +1220,44 @@ bool ContextAttribute::compoundItemExists(const std::string& compoundPath, orion
 
     for (unsigned int cIx = 0; cIx < current->childV.size(); ++cIx)
     {
-      if (current->childV[cIx]->name == compoundPathV[ix])
+      //
```

Comment:
You could use dbDotEncode()/dbDotDecode() functions, which hide the internal details of MongoDB codification (today we are using =, but in the future we could use some other). Something like this, I guess

```
if (dotEncode(current->childV[cIx]->name) == compoundPathV[ix])
```

or the other way around (not sure in this particular piece of code the encoding it is needed.

Reply:
Well, it's not about encoding, but about comparisons.
To keep this under control, what we could do is to to use defines for '.' and '=' and just change those defines if we (against all predictions) decide to change those chars.

Reply:
The defines already exist :-)
```
#define ESCAPE_1_DECODED  '.'
#define ESCAPE_1_ENCODED  '='
```

Not sure I like the names of the definitions though ...
`ESCAPE_1_DECODED` ... ?

If 'ESCAPE_1' was changed for 'DOT', it would be easier to understand, right?

Reply:
Fixed in 65ea8e296

Reply:
Regarding names for defines, ESCAPE_1 is probably due to a future extensibility that has not come. I mean, if in the future we need to do some other transformation, that would become ESCAPE_2 and so on. I'm open to suggestion for a name change proposal, althougth I don't like DOT very much, as the name should be independent of the actual char being tranformed.

With regards to comparisons, what I mean is that we can achieve the comparison in a very similar way we have right now (i.e. using `==` for strings) appliying an existing function (dbDotEncode/dbDotDecode) without needing any extra function for that (dotEqCompare in particular). 

However, I wouldn't like to block the PR for that, as the important thing now is the fix and accompaning .test. We can go as it is now and, once merged, I'd try to find a compact solution in the line I describe. If I fail, we can keep dotEqCompare. If I success, I'll remove dotEqCompare.

Reply:
Here we are again :-)
We will probably never agree on this kind of questions ...
The alternative solution that you propose, while less source code will be **a lot** slower in run-time.
My guess: 50 times slower.
As always, only for this code snippet, not for the entire execution of a request, of course.
But, why make the broker slower on purpose?
The function is very straightforward, we have other functions that are indefinitely more complex.

About the name of the define, yes, DOT is also not good. Let's just keep it as is, at least for now.

Reply:
Discusssed at skype. This PR will be merged as is. Later @fgalan will rollback that specif part to its original style.

NTC with regards this PR